specreduce/utils/utils.py

```diff
@@ -0,0 +1,209 @@
+    # list or a tuple to specify a range
+    if pixel is not None:
+        if pixel_range is not None:
+            raise ValueError('Both `pixel` and `pixel_range` can not be set'
+                             ' simultaneously.')
+        if isinstance(pixel, (int, float)):
+            pixels = np.array([int(pixel)])
+        elif np.all([isinstance(x, (int, float)) for x in pixel]):
+            pixels = np.array([int(x) for x in pixel])
+        else:
+            raise ValueError('`pixels` must be an integer, or list of integers '
+                             'to specify where the crossdisperion profile should '
+                             'be measured.')
+    else:  # range is specified
+        if pixel_range is None:
+            pixels = np.arange(0, ncols)
+        else:  # if not None, it should be a lower and upper bound
+            if len(pixel_range) != 2:
+                raise ValueError('`pixel_range` must be a tuple of integers.')
+            pixels = np.arange(min(pixel_range), max(pixel_range))
+
+    # now that we have all pixels that should be included in the profile, make
+    # sure that they are within image bounds.
+    # note: Should just warn instead and clip out out-of-bounds pixels, and only
+    # warn if there are none left?
+    if np.any(pixels < 0) or np.any(pixels > ncols - 1):
+        raise ValueError('Pixels chosen to measure cross dispersion profile are'
+                         ' out of image bounds.')
+
+    # now that we know which pixel(s) on the disp. axis we want to include
+    # figure out the range/window of pixels along the crossdisp axis to measure
+    # the profile
+    if width is None:  # if None, use all rows
+        width = nrows
+    elif isinstance(width, (float, int)):
+        width = int(width)
+    else:
+        raise ValueError('`width` must be an integer, or None to use all '
+                         'cross-dispersion pixels.')
+        width = int(width)
+
+    # rectify trace, if _align_along_trace is True and trace is not flat
+    aligned_trace = None
+    if align_along_trace:
+        if not isinstance(trace, FlatTrace):
+            # note: img was transposed according to `crossdisp_axis`: disp_axis will always be 1
+            aligned_trace = _align_along_trace(image, trace.trace,
+                                               disp_axis=1,
+                                               crossdisp_axis=0)
+
+            # new trace will be a flat trace at the center of the image
+            trace_pos = nrows / 2
+            trace = FlatTrace(aligned_trace, trace_pos)
+
+    # create a weight image based on the trace and 'width' to mask around trace
+
+    if width == nrows:
+        wimg = np.zeros(image.shape)
+    else:
+        wimg = _ap_weight_image(trace, width, disp_axis, crossdisp_axis, image.shape)
+        # invert mask to include, not exclude, pixels around trace
+        wimg = (1 - wimg).astype(int)
```

Comment:
would it be hard to get coverage for this scenario?

Reply:
done